pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -179,7 +179,8 @@ func (a *index) workloadEntryWorkloadBuilder(
 			log.Warnf("skipping workload entry %s/%s; DNS Address resolution is not yet implemented", wle.Namespace, wle.Name)
 		} // Else it is an empty address with network set, this is ok
 
-		w.WorkloadName, w.WorkloadType = wle.Name, workloadapi.WorkloadType_POD // XXX(shashankram): HACK to impersonate pod
+		w.WorkloadName = kubelabels.WorkloadNameFromWorkloadEntry(wle.Name, wle.Annotations, wle.Labels)
```

Comment:
Handling WorklogGroup labels or annotations (or both) - no problem.

My concern remains that this creates an annotation-based API for ambient - where injection is not required. I would rather have the pods that are not part of deployment/RS/etc use their name - and not require annotations.

An option is for the WorkloadGroup selector matching such lonely Pod.

Reply:
>  I would rather have the pods that are not part of deployment/RS/etc use their name

Then we will put in our prometheus metrics things like `application-756cd7dff4-ts8mk` and DOS prometheus.

Reply:
There is no current correlation between labels and annotation usage and ambient. There are plenty of existing usages of them, and no reason I see to not use them.